uint32 x_pos
uint32 y_pos
int16 cone_width
